File: exercice2/scala/src/main/scala/fr/devoxx/devops/logs/sql/SparkSQL1.scala
package fr.devoxx.devops.logs.sql

import fr.devoxx.devops.logs.ApacheAccessLog
import org.apache.spark.rdd.RDD
import org.apache.spark.sql.SQLContext

/* Les liens cassés */
case class SparkSQL1(rdd: RDD[String] , sqlContext: SQLContext) {

  def process: Long = {
    val dataFrame = sqlContext.createDataFrame(rdd.map(ApacheAccessLog.parse));
    dataFrame.registerTempTable("ApacheAccessLog");

    /*
    sqlContext.sql("select ...")
      .xxx
    */
    0L
  }
}

File: exercice2/java/src/main/java/fr/devoxx/devops/logs/sql/SparkSQL1.java
package fr.devoxx.devops.logs.sql;

import fr.devoxx.devops.logs.ApacheAccessLog;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.SQLContext;

/* Les liens cassés */
public class SparkSQL1 extends SparkSQL {

    public long process(JavaRDD<String> rdd, SQLContext sqlContext) {
        JavaRDD<ApacheAccessLog> accessLogs = rdd.map(ApacheAccessLog::parse);
        configure(sqlContext, accessLogs);

        /*
        return sqlContext.sql("select ...")
                .toJavaRDD()
                .map(...)
                .xxx();
        */
        return 0;
    }

}
